text/0788-bedrock-agentcore-memory-l2.md

```diff
@@ -0,0 +1,401 @@
+2. **Personalization**: Deliver consistent, personalized experiences based on user history
+3. **Flexible Strategies**: Choose from built-in or create custom extraction strategies
+4. **Secure Storage**: Automatic KMS encryption for sensitive information
+5. **Simple Integration**: Easy to add to existing agent applications
+6. **Managed Service**: No infrastructure to manage
+
+These L2 constructs eliminate the complexity of implementing persistent memory for AI agents.
+Developers can focus on building intelligent applications with context awareness.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+Amazon Bedrock AgentCore Memory addresses critical needs for AI agents:
+
+1. **Context Loss**: LLMs lack persistent memory across conversations
+2. **Personalization Needs**: Users expect agents to remember their preferences
+3. **Information Extraction**: Important facts need to be preserved for future use
+4. **Session Continuity**: Agents need to maintain context within conversations
+
+The L2 construct addresses these challenges by providing:
+
+- Managed short-term and long-term memory
+- Built-in extraction strategies
+- Secure storage with encryption
+- Simple configuration and deployment
+
+### Why should we _not_ do this?
+
+### What is the technical solution (design) of this feature?
+
+The L2 construct library is built using:
+
+1. **TypeScript with Projen**: Modern tooling for construct development
+2. **JSII**: Multi-language support (TypeScript, Python, Java, .NET)
+4. **Modular Architecture**: Memory and MemoryStrategy as separate constructs
+
+Key design principles:
+
+- **Flexibility**: Support for both built-in and custom strategies
+- **Sensible Defaults**: Production-ready configurations out of the box
+- **Extensibility**: Support for custom models and prompts
+- **Type Safety**: Strong typing for better developer experience
+
+## Interfaces
+
+The construct library provides comprehensive interfaces for Memory services:
+
+<will add link here to the public pull request containing all interfaces>
+
+### Is this a breaking change?
+
+No. This is a new construct library for Amazon Bedrock AgentCore Memory.
+It does not affect existing constructs.
+
+### What alternative solutions did you consider?
+
+### What are the drawbacks of this solution?
+
+### What is the high-level project plan?
+
+**Phase 1: RFC**:
+
+- Submit RFC proposal for creating the AgentCore Memory L2 construct
+- Design the initial interface and helper methods
+- Monitor the release of L1 constructs
+
+**Phase 2: Development**:
+
+- Create Memory module with strategy support
+- Implement built-in strategies
+- Add custom strategy support
+- Create comprehensive unit tests
+- Write comprehensive API documentation
+
+**Phase 3: Post-Launch**:
+
+- Publish launch blog and announcement posts
+- Regular updates to track AgentCore service changes
+- Add support for additional extraction strategies
+
+### Are there any open issues that need to be addressed later?
```

Comment:
As you mentioned earlier, the self-managed memory strategy have not had their Cloudformation API exposed yet. We should just mention this in the RFC so that we are aware of this in the future.